From e60200d25c21a3a0bde20d5ceaf1d619c791c975 Mon Sep 17 00:00:00 2001
From: Yuri Tolstov <ytolstov@marvell.com>
Date: Thu, 27 Dec 2018 10:13:09 +0530
Subject: [PATCH 0015/1921] octeontx2-af: Fix CGX promisc mode configuration

This patch fixes the promiscuous mode configration of CGX to allow
multicast packets as well.

Change-Id: Id10fc224e76a14d272e94c6233fd2db7609c49d9
Signed-off-by: Yuri Tolstov <ytolstov@marvell.com>
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 6d55e3d0b7ea..569f7532a5b8 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -258,8 +258,8 @@ void cgx_lmac_promisc_config(int cgx_id, int lmac_id, bool enable)
 	if (enable) {
 		/* Enable promiscuous mode on LMAC */
 		cfg = cgx_read(cgx, lmac_id, CGXX_CMRX_RX_DMAC_CTL0);
-		cfg &= ~(CGX_DMAC_CAM_ACCEPT | CGX_DMAC_MCAST_MODE);
-		cfg |= CGX_DMAC_BCAST_MODE;
+		cfg &= ~CGX_DMAC_CAM_ACCEPT;
+		cfg |= (CGX_DMAC_BCAST_MODE | CGX_DMAC_MCAST_MODE);
 		cgx_write(cgx, lmac_id, CGXX_CMRX_RX_DMAC_CTL0, cfg);
 
 		cfg = cgx_read(cgx, 0,
-- 
2.31.1

